lib/internal/modules/cjs/loader.js

```diff
@@ -619,6 +621,17 @@ Module._resolveFilename = function(request, parent, isMain, options) {
     return request;
   }
 
+  var filename;
+  // Don't use cache when:
+  // 1. require.resolve('you/request', {paths: []})
+  if (!(options && options.paths)) {
+    if (parent) {
+      filename = parent.resolvedFilenameCache &&
+      parent.resolvedFilenameCache[request];
+      if (filename) return filename;
+    }
+  }
+
```

Comment:
Instead of caching per parent would it be more benefitial to cache from the `path.dirname(parent.filename)`.
Something like:
```js
cacheKey =
      request + "\0" +
      fromPath + "\0" +
      (isMain ? "1" : "")
```

Reply:
Do you mean that using a global cache, which uses `request` + `parentDirName` + `isMain` as keys?
But this will not be deleted when people use `delete require.cache[filename]`. I'm a little concerned about that a global cache may cause some side effects when deleting cache.